src/go/docker-credential-none/cmd/root.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright © 2022 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	dockerconfig "github.com/docker/docker/cli/config"
+	"github.com/spf13/cobra"
+)
+
+const configFileName = "plaintext-credentials.config.json"
+
+type dockerConfigType map[string]interface{}
+
+type credType struct {
+	ServerURL string `json:"ServerURL"`
+	Username  string `json:"Username"`
+	Secret    string `json:"Secret"`
+}
+
+var configFile string
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use: "docker-credential-none",
+	Short: fmt.Sprintf(`Store docker credentials base64-encoded in ~/.docker/%s
+using the same format that docker uses when no credsStore field is specified in ~/.docker/config.json.
+This helper is intended for testing purposes, but will be used on Linux systems
+unless 'pass' and/or 'secretservice' is available.`,
+		configFileName),
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	err := rootCmd.Execute()
+	if err != nil {
+		os.Exit(1)
+	}
+}
+
+func init() {
+	configFile = filepath.Join(dockerconfig.Dir(), configFileName)
+}
+
+func getParsedConfig() (dockerConfigType, error) {
+	dockerConfig := make(dockerConfigType)
+	contents, err := ioutil.ReadFile(configFile)
+	if err != nil {
+		pathError, ok := err.(*os.PathError)
+		if ok && pathError.Err == syscall.ENOENT {
+			// Time to create a new config (or return no data)
+			return dockerConfig, nil
+		}
```

Comment:
Use [`errors.Is`](https://pkg.go.dev/errors#Is) instead:
```suggestion
		if errors.Is(err, syscall.ENOENT) {
			// Time to create a new config (or return no data)
			return dockerConfig, nil
		}
```